enterprise/backend/test/metabase_enterprise/advanced_permissions/api/general_test.clj

```diff
@@ -0,0 +1,68 @@
+(ns metabase-enterprise.advanced-permissions.api.general-test
+  (:require [clojure.test :refer :all]
+            [metabase.models :refer [PermissionsGroup]]
+            [metabase.models.permissions :as perms]
+            [metabase.models.permissions-group :as group]
+            [metabase.public-settings.premium-features-test :as premium-features-test]
+            [metabase.test :as mt]))
+
+(deftest general-permissions-test
+  (mt/with-temp* [PermissionsGroup [{group-id :id}]]
+    (testing "GET /api/ee/advanced-permissions/general/graph"
+      (premium-features-test/with-premium-features #{}
+        (testing "Should require a token with `:advanced-permissions`"
+          (is (= "This API endpoint is only enabled if you have a premium token with the :advanced-permissions feature."
+                 (mt/user-http-request :crowberto :get 402 "ee/advanced-permissions/general/graph")))))
+
+      (premium-features-test/with-premium-features #{:advanced-permissions}
+        (testing "have to be a superuser"
+          (is (= "You don't have permissions to do that."
+                 (mt/user-http-request :rasta :get 403 "ee/advanced-permissions/general/graph"))))
+
+        (testing "return general permissions for groups that has general permisions"
+          ;; TODO: remove this when subscription is granted by default for new group
+          (perms/grant-general-permissions! group-id :subscription)
+          (let [graph  (mt/user-http-request :crowberto :get 200 "ee/advanced-permissions/general/graph")
+                groups (:groups graph)]
+            (is (int? (:revision graph)))
+            (is (partial= {(mt/int->kw (:id (group/admin)))
+                           {:monitoring   "yes"
+                            :setting      "yes"
+                            :subscription "yes"}
+                           (mt/int->kw group-id)
+                           {:monitoring   "no"
+                            :setting      "no"
+                            :subscription "yes"}}
+                          groups)))))))
+
+  (mt/with-temp* [PermissionsGroup [{group-id :id}]]
+    (testing "PUT /api/ee/advanced-permissions/general/graph"
+      (let [current-graph (premium-features-test/with-premium-features #{:advanced-permissions}
+                            (mt/user-http-request :crowberto :get 200 "ee/advanced-permissions/general/graph"))
+            new-graph     (assoc-in current-graph [:groups (mt/int->kw group-id) :setting] "yes")]
+
+        (premium-features-test/with-premium-features #{}
+          (testing "Should require a token with `:advanced-permissions`"
```

Comment:
yea :) it's a bit misleading when reading from a test title POV. I'll change it

Reply:
I think if you read the full test title it will make more sense,  it'll be like this `PUT /api/ee/advanced-permissions/general/graph Should require a token with :advanced-permissions`

Basically saying this API will require a special feature to work.




FYI we use this title title in various place to check premium features:
https://github.com/metabase/metabase/blob/f6844db1a1779910ca5b89b9d22f2f215de2bc33/enterprise/backend/test/metabase_enterprise/audit_app/api/user_test.clj#L10